test/threads_exec.jl

```diff
@@ -1312,4 +1313,80 @@ end
         end
     end
 end
+
+@testset "task time counters" begin
+    @testset "enabled" begin
+        try
+            Base.task_timing(true)
+            start_time = time_ns()
+            t = Threads.@spawn peakflops()
+            wait(t)
+            end_time = time_ns()
+            wall_time_delta = end_time - start_time
+            @test t.is_timing_enabled
+            @test Base.task_cpu_time_ns(t) > 0
+            @test Base.task_wall_time_ns(t) > 0
+            @test Base.task_wall_time_ns(t) >= Base.task_cpu_time_ns(t)
+            @test wall_time_delta > Base.task_wall_time_ns(t)
+        finally
+            Base.task_timing(false)
+        end
+    end
+    @testset "disabled" begin
+        t = Threads.@spawn peakflops()
+        wait(t)
+        @test !t.is_timing_enabled
+        @test Base.task_cpu_time_ns(t) == 0
+        @test Base.task_wall_time_ns(t) == 0
+    end
+end
+
+@testset "task time counters: lots of spawns" begin
+    using Dates
+    try
+        Base.task_timing(true)
+        # create more tasks than we have threads.
+        # - all tasks must have: cpu time <= wall time
+        # - some tasks must have: cpu time < wall time
+        # - summing across all tasks we must have: total cpu time <= available cpu time
+        n_tasks = 2 * Threads.nthreads(:default)
+        cpu_times = Vector{UInt64}(undef, n_tasks)
+        wall_times = Vector{UInt64}(undef, n_tasks)
+        start_time = time_ns()
+        @sync begin
+            for i in 1:n_tasks
+                start_time_i = time_ns()
+                task_i = Threads.@spawn peakflops()
+                Threads.@spawn begin
+                    wait(task_i)
+                    end_time_i = time_ns()
+                    wall_time_delta_i = end_time_i - start_time_i
+                    cpu_times[$i] = cpu_time_i = Base.task_cpu_time_ns(task_i)
+                    wall_times[$i] = wall_time_i = Base.task_wall_time_ns(task_i)
+                    # task should have recorded some cpu-time and some wall-time
+                    @test cpu_time_i > 0
+                    @test wall_time_i > 0
+                    # task cpu-time cannot be greater than its wall-time
+                    @test wall_time_i >= cpu_time_i
+                    # task wall-time must be less than our manually measured wall-time
+                    # between calling `@spawn` and returning from `wait`.
+                    @test wall_time_delta_i > wall_time_i
```

Comment:
I don't think you're allowed to put `@test`s in spawned tasks, right? Or maybe you can, but the reporting is weird? I can't remember. Do you need to introduce a new `@testset` within this task?